tests/interlocked_int/interlocked_int.c

```diff
@@ -0,0 +1,855 @@
+//Copyright(c) Microsoft.All rights reserved.
+//Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifdef __cplusplus
+#include <cstdlib>
+#include <cstddef>
+#include <cstdint>
+#else
+#include <stdlib.h>
+#include <stddef.h>
+#include <stdint.h>
+#endif
+
+
+#include "testrunnerswitcher.h"
+
+static TEST_MUTEX_HANDLE g_testByTest;
+
+
+
+#include "azure_c_util/interlocked.h"
+
+
+BEGIN_TEST_SUITE(interlocked_int)
+
+
+TEST_SUITE_INITIALIZE(a)
+{
+    g_testByTest = TEST_MUTEX_CREATE();
+    ASSERT_IS_NOT_NULL(g_testByTest);
+
+    
+}
```

Comment:
Nitpick: some extra lines detected: this could make me think something is missing or has been removed when looking at the code. So to be neat would probably be nice to remove the extra lines.